android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+This document tries to provide a high level overview of some key differences (and similarities) of developing for Android versus developing for iOS. It is aimed to be a reference for people who are starting to or are considering switching from one mobile platform to another.
+
+# Language
+
+**TL;DR: Kotlin is more convenient, Swift is more powerful**
+
+## Common features and similarities (Kotlin and Swift)
+* Mainly object oriented (classes, inheritance, interface/protocol)
+* Statically typed
+* Type inference with option to specify explicit type
+* Type alias
+* Extensions
+* Generics
+* Optionals
+* Properties
+* Higher order functions
+* Default method parameters
```

Comment:
Relatedly, both also allow named parameters, though Swift requires parameter names to be specified almost always, whereas Kotlin only requires them for disambiguation.

Reply:
I'm opting to avoid mentioning syntax differences as these things can be learned pretty quickly and can't be missed.